src/ast/ast_validations.c

```diff
@@ -64,6 +64,43 @@ static bool _ValidateAllShortestPaths
 	return false;
 }
 
+// validate that shortestPaths is in a supported place
+static bool _ValidateShortestPaths
+(
+	const cypher_astnode_t *root // root to validate
+) {
+	ASSERT(root != NULL);
+
+	cypher_astnode_type_t t = cypher_astnode_type(root);
+	// if we found allShortestPaths in invalid parent return true
+	if(t == CYPHER_AST_SHORTEST_PATH &&
+	   cypher_ast_shortest_path_is_single(root)) {
+		return true;
+	}
+
+	// shortestPaths is invalid in the MATCH pattern
+	if(t == CYPHER_AST_MATCH) {
+		const cypher_astnode_t *pattern = cypher_ast_match_get_pattern(root);
+		return pattern != NULL && _ValidateShortestPaths(pattern);
+	}
```

Comment:
comment states that `shortestPaths` is INVALID in `MATCH` but we return true if there's a `pattern` and the `pattern` is `CYPHER_AST_SHORTEST_PATH`

Reply:
true is invalid in this function like the function above and it is invalid only on the pattern part not the predicate